queries.sql

```diff
@@ -103,4 +103,75 @@ WHERE owners.full_name ='Dean Winchester' AND animals.escape_attempts = 0;
 
 SELECT owners.full_name as OWNER , COUNT(animals.name) as count FROM
 owners JOIN animals ON owners.id = animals.owners_id GROUP BY owners.full_name
-ORDER BY count DESC LIMIT 1;
\ No newline at end of file
+ORDER BY count DESC LIMIT 1;
+
+/* Who was the last animal seen by William Tatcher? */
+
+SELECT vets.name AS vet, animals.name AS animal, 
+   visits.date_of_visits AS visited_date 
+   FROM visits 
+JOIN vets ON vets_id = vets.id 
+JOIN animals ON animals_id = animals.id 
+WHERE vets_id =
+  (SELECT id FROM vets WHERE name = 'William Tatcher') 
+ORDER BY date_of_visits DESC LIMIT 1;
+
+/* How many different animals did Stephanie Mendez see? */
+
+SELECT COUNT(DISTINCT animals_id) as no_of_different_animals FROM visits 
+JOIN vets ON visits.vets_id=vets.id 
+WHERE visits.vets_id=(SELECT id FROM vets WHERE name = 'Stephanie Mendez');
+
+/* List all vets and their specialties, including vets with no specialties. */
+
+SELECT vets.name AS vet, species.name AS specialized_in_species FROM vets 
+LEFT JOIN specializations ON vets.id=specializations.vets_id 
+LEFT JOIN species ON species_id=species.id;
+
+/* List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020. */
+
+SELECT animals.name as animal_name FROM visits  
+JOIN animals ON visits.animals_id = animals.id 
+WHERE visits.vets_id= (SELECT id FROM vets WHERE name = 'Stephanie Mendez') 
+AND visits.date_of_visits BETWEEN '2020-04-01' AND '2020-08-30';
+
+/* What animal has the most visits to vets? */
+
+ SELECT animals.name AS most_visited_animals, COUNT(*) AS number_of_visits FROM visits 
+ JOIN animals ON visits.animals_id=animals.id 
+ GROUP BY animals.name 
+ ORDER BY number_of_visits DESC LIMIT 1;
+
+/* Who was Maisy Smith's first visit? */
+
+SELECT animals.name as first_visited_animal FROM visits 
+JOIN vets ON visits.vets_id=vets.id
+JOIN animals ON visits.animals_id = animals.id
+WHERE vets.id=(SELECT id FROM vets WHERE name = 'Maisy Smith') 
```

Comment:
-  [ ]  [ OPTIONAL] Here you can use directly `WHERE vets.name = 'Maisy Smith'`, As this will give you same result.